core/src/main/java/brooklyn/entity/rebind/PeriodicDeltaChangeListener.java

```diff
@@ -348,13 +336,40 @@ private void addReferencedObjects(DeltaCollector deltaCollector) {
     }
     
     @VisibleForTesting
-    public void persistNow() {
-        if (!isActive()) {
+    public boolean persistNowSafely(boolean alreadyHasMutex) {
```

Comment:
Since the mutex is private it makes sense to expose to the outside only with alreadyHasMutex=false (i.e. a separate public metod which passes false), but not important.